### ISSUE TITLE
merge-patches: ensure local patch-queue branch exists

### ISSUE THREAD

**Issue by ktdreyer** [red-hat-storage/rhcephpkg#152]
When a local patch-queue branch does not exist, `merge-patches` behaves as if it was run with the `--force` option.

If there is a "patch-queue" branch in the origin remote, but it is missing locally, `merge-patches` should automatically set up the local one to match/track the origin one.

Like this:
```
git branch patch-queue/${DIST_GIT_BRANCH} -t origin/patch-queue/${DIST_GIT_BRANCH} || :
```
... and then continue on with the rest of the operations.

This will make the `merge-patches` behavior the same whether the local patch-queue branch is present or absent.

**Comment by ktdreyer** [comment]
We were seeing this problem today in Jenkins where I am not yet running with `--force`, but because I failed to set up a local tracking branch, `merge-patches` effectively just resets anyway, and then Jenkins fails much later on when it tries to push the diverging patch-queue branch back to the remote.

**Comment by ktdreyer** [comment]
Longer term, we may want to just rename "merge-patches" to "reset-patches" and reduce some of the complexity here.

**Comment by ktdreyer** [comment]
Fixed in v1.7.0
